source/ShipyardPanel.cpp

```diff
@@ -131,7 +131,7 @@ int ShipyardPanel::DrawPlayerShipInfo(const Point &point)
 bool ShipyardPanel::HasItem(const string &name) const
 {
 	const Ship *ship = GameData::Ships().Get(name);
-	return shipyard.Has(ship);
+	return shipyard.Has(ship) || player.Stock(ship) > 0;
```

Comment:
can you explain to me how the amount would be negative in the stock?

Reply:
Sure!  Buy stuff. ;)  When the player buys things, stock gets reduced.  So, for stuff the shop sells, it starts at zero and then goes negative as they buy stuff.  We only need to worry about it when it's positive (the player has net sold it to the shop).

We have to track it for stuff the shop sells because of license restricted items - we want the player to be able to buy them back even if they don't have a license for them, just like with outfits.